service/bfg/bfg.go

```diff
@@ -1766,3 +1786,122 @@ func (s *Server) Run(pctx context.Context) error {
 
 	return err
 }
+
+// remoteIP returns the remote client IP address for the http request.
+func (s *Server) remoteIP(req *http.Request) string {
+	remoteAddr, _, err := net.SplitHostPort(req.RemoteAddr)
+	if err != nil {
+		return ""
```

Comment:
should we at least log this error out?

Reply:
We could add a log here, but this should never happen.